pages/apis/graphql/cookbooks/pipeline_templates.md

```diff
@@ -0,0 +1,145 @@
+
+```graphql
+query GetPipelineTemplates {
+  organization(slug: "organization-slug") {
+    pipelineTemplates(first: 10) {
+      edges {
+        node {
+          id
+          uuid
+          name
+          description
+          configuration
+          available
+        }
+      }
+    }
+  }
+}
+```
+
+## Get a pipeline template
+
+Get information on a pipeline template, specifying the pipeline templates' UUID as the `uuid` argument of the `pipelineTemplate` query:
+
+```graphql
+query GetPipelineTemplate {
+  pipelineTemplate(uuid: "pipeline-template-uuid") {
+    id
+    uuid
+    name
+    description
+    configuration
+    available
+  }
+}
+```
+
+## Create a pipeline template
+
+Create a pipeline template for an organization using the `pipelineTemplateCreate` mutation:
+
+```graphql
+mutation CreatePipelineTemplate {
+  pipelineTemplateCreate(input: {
+    organizationId: "organization-id",
+    name: "template name",
+    description: "it does a thing",
+    configuration: "steps:\n  - command: deploy.sh",
+    available: false
+  }) {
+    pipelineTemplate {
+      id
+      uuid
+      name
+      description
+      configuration
+      available
+    }
+  }
+}
+```
+
+## Updating a pipeline template
+
+Update a pipeline template on an organization using the `pipelineTemplateUpdate` mutation, specifying the ID for organization and pipeline template:
+
+```graphql
+mutation UpdatePipelineTemplate {
+  pipelineTemplateUpdate(input: {
+    organizationId: "organization-id",
+    id: "pipeline-template-id",
+    configuration: "steps:\n - comand: updated_steps.sh"
+    available: true
+  }) {
+    pipelineTemplate {
+      id
+      uuid
+      name
+      description
+      configuration
+      available
+    }
+  }
+}
+```
+
+## Deleting a pipeline template
+
+Delete a pipeline template using the `pipelineTemplateDelete` mutation, specifying the ID for organization and pipeline template:
+
+```graphql
+mutation DeletePipelineTemplate {
+  pipelineTemplateDelete(input: {
+    organizationId: "organization-id",
+    id: "pipeline-template-id"
+  }) {
+    deletedPipelineTemplateId
+  }
+}
+```
+
+## Managing pipeline template assignment on a pipeline
+
+Admins and users with manage pipeline permissions can assign a pipeline template to a pipeline using the `pipelineUpdate` mutation:
+
+```graphql
+mutation AssignPipelineTemplate {
+  pipelineUpdate(input: {
+    id: 'pipeline-id'
+    pipelineTemplateId: 'pipeline-template-id'
+  }) {
+    pipeline {
+      id
+      name
+      pipelineTemplateId
+    }
+  }
+}
+```
+
+Conversely, pipeline templates can be removed from a pipeline by specifying `pipelineTemplateId` as `null` in the mutation input:
+
+```graphql
+mutation UnassignPipelineTemplate {
+  pipelineUpdate(input: {
+    id: 'pipeline-id'
+    pipelineTemplateId: null
+  }) {
+    pipeline {
+      id
+      name
+      pipelineTemplateId
+    }
+  }
+}
+```
```

Comment:
Seems like this should be using double quotes (syntax error is highlighted in the generated docs too):

<img width="681" alt="image" src="https://github.com/buildkite/docs/assets/673828/dacdb34b-5c4b-4d76-aa0c-6061b940ed71">

````suggestion
## Managing pipeline template assignment on a pipeline

Admins and users with manage pipeline permissions can assign a pipeline template to a pipeline using the `pipelineUpdate` mutation:

```graphql
mutation AssignPipelineTemplate {
  pipelineUpdate(input: {
    id: "pipeline-id"
    pipelineTemplateId: "pipeline-template-id"
  }) {
    pipeline {
      id
      name
      pipelineTemplateId
    }
  }
}
```

Conversely, pipeline templates can be removed from a pipeline by specifying `pipelineTemplateId` as `null` in the mutation input:

```graphql
mutation UnassignPipelineTemplate {
  pipelineUpdate(input: {
    id: "pipeline-id"
    pipelineTemplateId: null
  }) {
    pipeline {
      id
      name
      pipelineTemplateId
    }
  }
}
```
````

Reply:
Ah great pick up @mhaylock, fixed up the syntax in [8f472e4](https://github.com/buildkite/docs/pull/2479/commits/8f472e4e12307e1ba5011e4eb4d0c9adb608d6d2)